### ISSUE TITLE
Run command not working

### ISSUE THREAD

**Issue by AlanTJohnstone** [spyder-ide/spyder#21089]
### What steps will reproduce the problem?

I am using Spyder 4.2.1  on RPI4 using Python 3.9
Hitting run (or the green arrow) does nothing.
I can get the program to run by highlighting it all and choosing 'Run Selection'
F5 also does not work

## Versions

* Spyder version: 4.2.1 
* Python version: 3.9.2
* Qt version: 5.15.2
* PyQt5 version: 5.15.2
* Operating System: Linux 6.1.21-v8+

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0            :  1.4.0 (OK)
chardet >=2.0.0                 :  4.0.0 (OK)
cloudpickle >=0.5.0             :  1.6.0 (OK)
diff_match_patch >=20181111     :  20200713 (OK)
intervaltree >=3.0.2            :  3.0.2 (OK)
IPython >=7.6.0                 :  7.20.0 (OK)
jedi >=0.17.2                   :  0.18.0 (OK)
jsonschema >=3.2.0              :  3.2.0 (OK)
keyring >=17.0.0                :  22.0.1 (OK)
nbconvert >=4.0                 :  5.6.1 (OK)
numpydoc >=0.6.0                :  1.1.0 (OK)
parso >=0.7.0                   :  0.8.1 (OK)
pexpect >=4.4.0                 :  4.8.0 (OK)
pickleshare >=0.4               :  0.7.5 (OK)
psutil >=5.3                    :  5.8.0 (OK)
pygments >=2.0                  :  2.7.1 (OK)
pylint >=1.0                    :  2.7.2 (OK)
pyls >=0.36.2;<1.0.0            :  0.36.2 (OK)
pyls_black >=0.4.6              :  0.4.6 (OK)
pyls_spyder >=0.3.0             :  0.3.0 (OK)
qdarkstyle >=2.8                :  2.8.1 (OK)
qtawesome >=0.5.7               :  1.0.2 (OK)
qtconsole >=5.0.1               :  5.0.2 (OK)
qtpy >=1.5.0                    :  1.9.0 (OK)
setuptools >=39.0.0             :  52.0.0 (OK)
sphinx >=0.6.6                  :  3.4.3 (OK)
spyder_kernels >=1.10.1;<1.11.0 :  1.10.2 (OK)
textdistance >=4.2.0            :  4.2.0 (OK)
three_merge >=0.1.1             :  0.1.1 (OK)
watchdog >=0.10.3               :  1.0.2 (OK)
xdg >=0.26                      :  0.27 (OK)
zmq >=17                        :  20.0.0 (OK)

# Optional:
cython >=0.21                   :  None (NOK)
matplotlib >=2.0.0              :  3.7.1 (OK)
numpy >=1.7                     :  1.24.2 (OK)
pandas >=1.1.1                  :  1.5.3 (OK)
scipy >=0.17.0                  :  1.10.1 (OK)
sympy >=0.7.3                   :  None (NOK)
```

**Comment by dalthviz** [comment]
Hi @AlanTJohnstone thank you for the feedback! Is there any reason why you are still using Spyder 4.2.1? The latest Spyder version available is 5.4.3. Spyder 4.2.1 is almost 3 years old! If you still need to use Spyder 4.x, maybe try updating to the the last Spyder 4.x release (version 4.2.5), if not I would suggest you to update to Spyder 5.4.3 and check again.

I think there is not much we can do other than to suggest you to update to a more recent Spyder version :/ but let us know if you can update and if that helps!

**Comment by AlanTJohnstone** [comment]
Thank you for your quick response.
The reason why I am using 4.2.1 is simple. That is what I got when I downloaded  via APT. When I open the program I get a popup suggesting I upgrade but when I followed the link it described the upgrade as ‘experimental’ for linux. So I decided to work with something tried and tested.

Tomorrow I will upgrade and see, but I am sure the problem is not version related. I am sure you did not release a stable version with that did not run a program!!.

Alan
